### PR TITLE
test: add test on client redirect

### PR DESCRIPTION
**Pull request by smira**
This matches PR https://github.com/siderolabs/discovery-client/pull/4.

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -2,17 +2,19 @@ module github.com/siderolabs/discovery-service
 
 go 1.19
 
+replace github.com/siderolabs/discovery-client => github.com/smira/discovery-client v0.0.0-20220915115237-230f317a8e6e
```
Comment (frezbo) [inline go.mod:5]: i guess we can remove this now?

Comment (smira) [reply go.mod:5]: once the other PR is merged, I will do that now

**Review by frezbo — APPROVED** [review]
:cool:

**Pull request merged**